# Imports

In [17]:
import os
import pandas as pd
from helpers import *

pd.set_option("display.max_colwidth", None)

# Wave Data Extraction
Parse raw data sets to extract questions relevant to the research topic. Relevant questions ask respondants about either military aid for Ukraine or economic sanctions on Russia.
Additionally extract fieldwork timelines during which respondants were surveyed.

In [25]:
collected = []
for file in os.listdir(EB_BASE_PATH):
    if file.endswith("xlsx"):
        collected.append(parse_doc(file))
collected.sort(key=lambda k: k["wave_id"])
df = pd.DataFrame(collected)
df

,eb_n,wave_id,fw_start,fw_end,season,questions,q_ids
0,eb97,97.5,2022-06-17,2022-07-17,Summer 2022,"[QE2.1. The EU has taken a series of actions as a response to Russia’s invasion in Ukraine. To what extent you agree or disagree with each of these actions taken. :-Imposing economic sanctions on Russian government, companies and individuals, QE2.3. The EU has taken a series of actions as a response to Russia’s invasion in Ukraine. To what extent you agree or disagree with each of these actions taken. :-Financing supply and delivery of military equipment to Ukraine]","[QE2_1, QE2_3]"
1,eb98,98.2,2023-01-12,2023-02-06,Spring 2023,"[QE2.1. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Imposing economic sanctions on Russian government, companies and individuals, QE2.3. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Financing the purchase and supply of military equipment to Ukraine]","[QE2_1, QE2_3]"
2,eb99,99.4,2023-05-31,2023-06-25,Spring 2023,"[QE2.1. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Imposing economic sanctions on Russian government, companies and individuals, QE2.3. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Financing the purchase and supply of military equipment to Ukraine]","[QE2_1, QE2_3]"
3,eb100,100.2,2023-10-23,2023-11-17,Fall 2023,"[QD2.1. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Imposing economic sanctions on Russian government, companies and individuals, QD2.3. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Financing the purchase and supply of military equipment to Ukraine]","[QD2_1, QD2_3]"
4,eb101,101.3,2024-04-02,2024-05-09,Spring 2024,"[QD2.1. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Imposing economic sanctions on Russian government, companies and individuals, QD2.2. The EU has taken a series of actions as a response to Russia's invasion of Ukraine. To what extent you agree or disagree with each of these actions taken. :-Financing the purchase and supply of military equipment to Ukraine]","[QD2_1, QD2_2]"
5,eb102,102.2,2024-10-10,2024-11-05,Fall 2024,"[QD2.1. The EU has taken a series of actions as a response to Russia’s invasion of Ukraine. To what extent you agree or disagree with each of these actions taken?:-Imposing economic sanctions on Russian government, companies and individuals, QD2.2. The EU has taken a series of actions as a response to Russia’s invasion of Ukraine. To what extent you agree or disagree with each of these actions taken?:-Financing the purchase and supply of military equipment to Ukraine]","[QD2_1, QD2_2]"
6,eb103,103.3,2025-03-26,2025-04-22,Spring 2025,"[QD2.1. The EU has taken a series of actions as a response to Russia’s invasion of Ukraine. To what extent you agree or disagree with each of these actions taken.:-Imposing economic sanctions on Russian government, companies and individuals, QD2.2. The EU has taken a series of actions as a response to Russia’s invasion of Ukraine. To what extent you agree or disagree with each of these actions taken.:-Financing the purchase and supply of military equipment to Ukraine]","[QD2_1, QD2_2]"
7,eb104,104.1,2025-10-09,2025-11-05,Fall 2025,"[QD2.1. The EU has taken a series of actions as a response to Russia’s invasion of Ukraine. To what extent you agree or disa

In [27]:
qs = pd.DataFrame(df["questions"].tolist(), index=df.index)
qs.columns = ["q1", "q2"]
qs["q1"] = qs["q1"].apply(lambda v: f"{v.split(" ", 1)[0]} {v.split(":-")[-1]}")
qs["q2"] = qs["q2"].apply(lambda v: f"{v.split(" ", 1)[0]} {v.split(":-")[-1]}")
questions = df.join(qs).drop(columns=["questions", "q_ids"])
questions

,eb_n,wave_id,fw_start,fw_end,season,q1,q2
0,eb97,97.5,2022-06-17,2022-07-17,Summer 2022,"QE2.1. Imposing economic sanctions on Russian government, companies and individuals",QE2.3. Financing supply and delivery of military equipment to Ukraine
1,eb98,98.2,2023-01-12,2023-02-06,Spring 2023,"QE2.1. Imposing economic sanctions on Russian government, companies and individuals",QE2.3. Financing the purchase and supply of military equipment to Ukraine
2,eb99,99.4,2023-05-31,2023-06-25,Spring 2023,"QE2.1. Imposing economic sanctions on Russian government, companies and individuals",QE2.3. Financing the purchase and supply of military equipment to Ukraine
3,eb100,100.2,2023-10-23,2023-11-17,Fall 2023,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals",QD2.3. Financing the purchase and supply of military equipment to Ukraine
4,eb101,101.3,2024-04-02,2024-05-09,Spring 2024,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals",QD2.2. Financing the purchase and supply of military equipment to Ukraine
5,eb102,102.2,2024-10-10,2024-11-05,Fall 2024,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals",QD2.2. Financing the purchase and supply of military equipment to Ukraine
6,eb103,103.3,2025-03-26,2025-04-22,Spring 2025,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals",QD2.2. Financing the purchase and supply of military equipment to Ukraine
7,eb104,104.1,2025-10-09,2025-11-05,Fall 2025,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals",QD2.2. Financing the purchase and supply of military equipment to Ukraine
8,eb105,105.2,2026-03-12,2026-04-05,Spring 2026,"QD2.1. Imposing economic sanctions on Russian government, companies and individuals (including use of immobilised Russian assets to finance support for Ukraine)",QD2.2. Financing the purchase and supply of military equipment to Ukraine


Looking at the extracted questions from each wave, we can see almost no change in the wording of the questions other than QE2.3 in wave 97.5 and QD2.1 in wave 105.2. However, given that the deviations in wording are relatively minor - 97.5-QE2.3 implies the same thing as 98.2-QE2.3 and 105.2-QD2.1 clarifies the scope of the sanctions - a consistent analysis over time can still be made.